pilot/cmd/pilot-discovery/main.go

```diff
@@ -152,6 +184,28 @@ func init() {
 	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.DiscoveryOptions.EnableCaching, "discoveryCache", true,
 		"Enable caching discovery service responses")
 
+	// Chiron options
+	var chironServiceNames string
+	var chironServiceNamespaces string
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.Chiron.EnableChiron, "enableChiron", false,
+		"Enable Chiron")
+	// Specifies the file path to k8s CA certificate.
+	// The default value is configured based on https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/:
+	// the CA certificate bundle is automatically mounted into pods using the default
+	// service account at the path /var/run/secrets/kubernetes.io/serviceaccount/ca.crt.
+	discoveryCmd.PersistentFlags().StringVar(&serverArgs.Chiron.K8sCaCertFile, "k8sCaCertFile", "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
```

Comment:
Those things should go under mesh config - the agreement in EnvWG was to use the mesh config in API for configuration and stop adding CLI ( and associated values.yaml ). See my other comment.

This config in particular doesn't seem to be needed - this is a well-known path defined by k8s, so can be a constant.
If file is not found, we can use the value from the KUBECONFIG ( see my change in costinm/istio-vm )

Reply:
Done: moved to mesh config.
The parameter for configuring the path to the public certificate bundle file of the k8s signing CA is needed because the k8s CA signing cert may be customized through --cluster-signing-cert-file and this parameter needs to be consistent with the k8s CA signing certificate.